packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -529,6 +529,43 @@ export function isTaggedTelemetryPropertyValue(x: any): x is ITaggedTelemetryPro
 
 export const isILoggingError = (x: any): x is ILoggingError => typeof x?.getTelemetryProperties === "function";
 
+/**
+ * Read-Write Logging Error.  Not exported.
+ * This type alias includes addTelemetryProperties, and applies to objects even if not instanceof LoggingError\
+ */
+type RwLoggingError = LoggingError;
+
+/** Type guard for RwLoggingError.  Not exported. */
+const isRwLoggingError = (x: any): x is RwLoggingError =>
+    typeof x?.addTelemetryProperties === "function" && isILoggingError(x);
+
+/**
+ * Annotate the given error object with the given logging props
+ * @returns The same error object passed in if possible, with telemetry props functionality mixed in
+ */
+export function annotateError(
+    error: unknown,
+    props: ITelemetryProperties,
+): ILoggingError {
+    if (isRwLoggingError(error)) {
+        error.addTelemetryProperties(props);
+        return error;
+    }
+
+    if (isRegularObject(error)) {
+        // Even though it's not exposed, fully implement IRwLoggingError for subsequent calls to annotateError
+        const loggingError = error as RwLoggingError;
+
+        const propsForError = {...props};
+        loggingError.getTelemetryProperties = () => propsForError;
+        loggingError.addTelemetryProperties = (p: ITelemetryProperties) => { Object.assign(propsForError, p); };
```

Comment:
annotations will override existing properties which i'm not sure is correct

Reply:
Actually I'm going to keep it as-is for now, to match `LoggingError.addTelemetryProperties`.  I will submit a follow-up PR to flip them both.